src/components/sections/EditorSection.tsx

```diff
@@ -0,0 +1,45 @@
+import { useAlert, useDataQuery } from '@dhis2/app-runtime'
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import Editor from './Editor'
+
+const EditorSection = ({ query }) => {
+    const { key, namespace, store } = useParams()
+    const { show: showError } = useAlert('An error fetching this data', {
+        critical: true,
+    })
+
+    const { data, loading, refetch } = useDataQuery(query, {
```

Comment:
This is not really a comment for the implementer/implantation. But I think in general it's kind of sad to refactor an app away from something like `useSwr` or `react-query` to `useDataQuery` that is in general inferior in terms of caching data, correctness and DX. I would much rather prefer to use `useQuery` directly. But yeah, this is a controversial topic.

Reply:
This is a conversation worth having. I focused on the use of an in-house tool, i.e. the `app-runtime` for this.   However, if there is something better, it can and should be considered.

Reply:
@Birkbjo I was just looking at your doc about app-runtime, not so controversial.  However this app wasn't previously using `useSwr` or `react-query`...?  It was using Redux and [d2](https://github.com/dhis2/datastore-app/blob/master/src/utils/api.js).

This app is pretty simple so for now introducing a separate data state management library is unnecessary.  We'd rather focus on improving our tools where they have weaknesses.

Reply:
> However this app wasn't previously using useSwr or react-query...? It was using Redux and [d2](https://github.com/dhis2/datastore-app/blob/master/src/utils/api.js).

Ahh... Sorry I totally confused this with `app-hub`! You're right that moving away from d2 is definitely a good change. 
And I realize that my comment was a bit thin on its own without a proper alternative or solution. But I just wanted to bring it up that `useDataQuery` forces you to write non-ideal code (in my opinion). 

But yeah this app is hopefully simple enough that other tools might not be necessary.